src/dapi/src/lts/loaddict.c

```diff
@@ -469,19 +469,19 @@ restart:if ( *dict_siz > 0 )
 
 #endif
 #ifdef CHEESY_DICT_COMPRESSION
-	if (dict_fc_entry!=NULL)
-	{
-		fc_entry = (int *)(((int)*dicMapStartAddr) + 12); //start the index buffer at start address + 8 bytes
+		if (dict_fc_entry!=NULL)
+		{
+			fc_entry = (int *)(((int)*dicMapStartAddr) + 12); //start the index buffer at start address + 8 bytes
 
-		dict_index_buffer = (int *)(((long)*dicMapStartAddr) + 12 + fc_entries*4); //start the index buffer at start address + 8 bytes
-	}
+			dict_index_buffer = (int *)(((long)*dicMapStartAddr) + 12 + fc_entries*4); //start the index buffer at start address + 8 bytes
+		}
 	else
 #endif
-	{
-		dict_index_buffer = (int *)(((long)*dicMapStartAddr) + 8); //start the index buffer at start address + 8 bytes
-	}
+		{
+			dict_index_buffer = ((((QWORD)*dicMapStartAddr) + 8)); //start the index buffer at start address + 8 bytes

```

Comment:
I think the (`int *`) cast is actually needed here, as the dict_index_buffer is a S32*. Without the cast I get a new compiler warning on linux:
loaddict.c:481:22: warning: assignment to ‘S32 *’ {aka ‘int *’} from ‘long long unsigned int’ makes pointer from integer without a cast -Wint-conversion]

The QWORD change should be ok (windows is strange in the regard that long is 32bit on 64bit platforms).